src/com/phonebookservice/server/FileDataAccessAdapter.java

```diff
@@ -1,13 +1,54 @@
 package com.phonebookservice.server;
 
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.Writer;
+import java.util.List;
+
+import org.apache.commons.io.FileUtils;
+
+import com.google.common.collect.BiMap;
+import com.google.common.collect.HashBiMap;
+import com.phonebookservice.exception.InternalServerException;
 import com.phonebookservice.model.Contact;
+import com.phonebookservice.util.CollectionUtility;
+import com.phonebookservice.util.MyArrayList;
 
 /**
  * file data access adapter.
  *
  * @author Marwa Saleh
  */
-public class FileDataAccessAdapter implements IDataAccessAdapter<Contact> {
+public final class FileDataAccessAdapter
+        implements IDataAccessAdapter<Contact> {
+    private static FileDataAccessAdapter singleton;
+    private static final String FILE_NAME = "D:\\phoneBookService\\src\\com\\"
```

Comment:
adding the whole path is no acceptable, that what caused the tests to fail (d does not exist here)